yt_dlp/extractor/iprima.py

```diff
@@ -134,10 +134,17 @@ def _real_extract(self, url):
         ), webpage, 'real id', group='id', default=None)
 
         if not video_id:
-            nuxt_data = self._search_nuxt_data(webpage, video_id, traverse='data')
+            nuxt_data = self._search_nuxt_data(webpage, video_id, traverse='data', fatal=False)
             video_id = traverse_obj(
                 nuxt_data, (..., 'content', 'additionals', 'videoPlayId', {str}), get_all=False)
 
+        if not video_id:
+            nuxt_data = self._search_json(
+                r'<script[^>]+\bid=["\']__NUXT_DATA__["\'][^>]*>',
+                webpage, 'nuxt data', None, end_pattern=r'</script>', contains_pattern=r'(?s:.+?)')
```

Comment:
Does this work?
```suggestion
            nuxt_data = self._search_json(
                r'<script[^>]+\bid=["\']__NUXT_DATA__["\'][^>]*>',
                webpage, 'nuxt data', None, end_pattern=r'</script>', contains_pattern=r'\[(?s:.+)\]')
```

Reply:
> Yes, it does, but I'd like to keep the regex non-greedy because if there were multiple scripts on the same line, the contents of both (+ the mark-up) could be matched, with the end_pattern matching the end of the second one.

The greedy search is desired; `_search_json` calls `_parse_json` with `ignore_extra=True`, which has the `LenientJSONDecoder` truncate wherever `json.loads` throws. It also avoids catastrophic backtracking

> Also I do not particularly think we need to specify the array brackets - maybe it could even be useful if they changed it to an object and the code happened to work (though unlikely

The traversal would not work if they did this, so I think as long as my suggestion works currently, it is what we should use

Reply:
Alright, did not know about LenientJSONDecoder, will trust that it works.

Currently your changes work, I've tested it before committing.

I am aware of catastrophic backtracking but with that specific regex I cannot think of an input that would be problematic, even a deliberately malicious one. Maybe my lack of imagination, if you can think of one, I would love an example (just curious).

> The traversal would not work if they did this, so I think as long as my suggestion works currently, it is what we should use

I mean it would if the first value in the object that met the condition was our video id, right? Though not sure about iteration order here (not sure if it is the original one or a random one).

Reply:
> I mean it would if the first value in the object that met the condition was our video id, right?

Yeah, you are right, my bad. I still think it's better to be specific here, though. If the site does change the data from array to object, either 1. it would be a quick and easy fix or 2. they've changed other things as well and it would be helpful to have `_search_json` throw